PowerFGT/Public/cmdb/system/zone.ps1

```diff
@@ -95,6 +95,190 @@ function Get-FGTSystemZone {
         $response.results
     }
 
+    End {
+    }
+}
+
+function Add-FGTSystemZone {
+
+    <#
+        .SYNOPSIS
+        Add a zone
+
+        .DESCRIPTION
+        Add a zone
+
+        .EXAMPLE
+        Add-FGTSystemZone -name PowerFGT -intrazone deny -interfaces port5,port6
+
+        Add a zone named PowerFGT with intra-zone traffic blocked and with port5 and port6 in this zone
+    #>
+
+    [CmdletBinding(DefaultParameterSetName = "default")]
+    Param(
+        [Parameter (Mandatory = $false, Position = 1)]
+        [string]$name,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('allow','deny')]
+        [string]$intrazone,
+        [Parameter(Mandatory = $false)]
+        [string[]]$interfaces,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $zone = new-Object -TypeName PSObject
+        if ( $PsBoundParameters.ContainsKey('name') ) {
+            $zone | add-member -name "name" -membertype NoteProperty -Value $name
+        }
+        Else{
+            Throw "You need to specifiy a name"
+        }
+
+        $get_zone_name = Get-FGTSystemZone -name $name
+
+        If($null -ne $get_zone_name)
+        {
+            Throw "Already a zone using the same name"
+        }
+
+        if ( $PsBoundParameters.ContainsKey('interfaces') ) {
+            $ports = @()
+            foreach ( $member in $interfaces ) {
+                $get_interface = Get-FGTSystemInterface -name $member
```

Comment:
Missing connection parameter and don’t check for the moment, I will plan a cache for avoid to check for each interface (can be very slow...)